Sources/NIO/System.swift

```diff
@@ -535,6 +556,50 @@ internal extension Posix {
     }
 }
 
+internal extension Posix {
+    static func cmsgFirstHeader(inside msghdr: UnsafePointer<msghdr>?) -> UnsafeMutablePointer<cmsghdr>? {
+        return sysCmsgFirstHdr(msghdr)
+    }
+    
+    static func cmsgNextHeader(inside msghdr: UnsafeMutablePointer<msghdr>?, from: UnsafeMutablePointer<cmsghdr>?)
+        -> UnsafeMutablePointer<cmsghdr>? {
+        return sysCmsgNxtHdr(msghdr, from)
+    }
+    
+    static func cmsgData(for header: UnsafePointer<cmsghdr>?) -> UnsafeRawBufferPointer? {
+        let dataPointer = sysCmsgData(header)
+        if let header = header {
```

Comment:
This `if let` is really in error: we aren't allowed to call `sysCmsgData` with a nil pointer. Let's remove the nullability in the API to reflect that, and save ourselves the branch.